marklogic-data-hub-central/ui/src/components/entities/create-edit-step-dialog/create-edit-step-dialog.test.tsx

```diff
@@ -63,9 +64,39 @@ describe('Create Edit Step Dialog component', () => {
 
         fireEvent.change(nameInput, { target: {value: ''}});
         expect(getByText('Name is required')).toBeInTheDocument();
-        expect(saveButton).toBeDisabled();
+        expect(saveButton).toBeEnabled();
+    });
+
+    test('Verify clicking "Save" with no collection shows error', async () => {
+      const { getByText, getByLabelText, getByPlaceholderText } = render(<CreateEditStepDialog {...data.newMerging} />);
+
+      fireEvent.change(getByPlaceholderText('Enter name'), { target: {value: ''}});
+  
+      fireEvent.click(getByText('Save'));
+  
+      // both messages should show when both boxes are empty
+      expect(getByText('Name is required')).toBeInTheDocument();
```

Comment:
Let's verify all the mandatory fields in one test only to avoid unnecessary re-renders of CreateEditStepDialog component in a separate test, wherever possible. We can describe the test heading may be like "Verify clicking 'Save' with no value for mandatory fields show error" and then just do your assertions like below:-
```
Set a non-null value for name.
verify if error for collection field appears on clicking "save".
Reset name value to empty('') and set valid collection value.
Click on save button and verify if error for name field is available in the document or not.
```

Reply:
Squashed all the new tests together.